src/components/controlled-input/controlled-input.ts

```diff
@@ -4,35 +4,51 @@ import { InputProps } from '../input/input';
 import { validateInput, ValidationRule } from '../../core/validator';
 
 interface ControlledInputProps extends InputProps {
-  validationRule: ValidationRule,
+  validationRule: ValidationRule;
 }
 
-export default class ControlledInput extends Block {
-  public static componentName = 'ControlledInput';
+export default class ControlledInput extends Block<ControlledInputProps> {
+  public static override componentName = 'ControlledInput';
 
   constructor({ validationRule, ...props }: ControlledInputProps) {
     super({
       ...props,
+      validationRule,
       onBlur: (e: FocusEvent) => {
         const { value } = e.target as HTMLInputElement;
         const validationResult = validateInput({
           rule: validationRule,
           value,
         });
-        this.refs.error.setProps({ text: validationResult });
+        const { error } = this.refs;
+        if (error) {
+          error.setProps({ text: validationResult });
+        }
```

Comment:
Можно короче записать
```suggestion
  error?.setProps({ text: validationResult });
```